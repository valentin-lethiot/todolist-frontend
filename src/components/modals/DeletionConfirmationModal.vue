<script setup lang="ts">
import { useI18n } from 'vue-i18n'
const { t } = useI18n()

type Props = {
  isOpen: boolean
}
const { isOpen } = defineProps<Props>()

const emit = defineEmits(['close', 'confirm'])

const emitClose = () => emit('close')
const emitConfirm = () => emit('confirm')
</script>

<template>
  <div v-show="isOpen" class="absolute w-screen h-screen top-0 left-0 bg-gray-900/40 z-60">
    <div class="w-full h-full flex flex-col justify-center items-center p-8">
      <div class="bg-gray-800 rounded-lg p-4 shadow-xl w-1/2">
        <div class="flex flex-col w-full h-full gap-4">
          <div class="h-1/2 text-xl font-bold flex justify-between items-start">
            <p>{{ t('sureToDelete') }}</p>
          </div>

          <div class="h-1/2 w-full flex gap-4 items-center justify-end">
            <button
              id="cancelButton"
              @click="emitClose"
              class="bg-gray-700 hover:bg-gray-600 rounded-lg text-lg font-bold p-2 text-white w-1/2 hover:cursor-pointer"
            >
              {{ t('actions.cancel') }}
            </button>
            <button
              id="confirmButton"
              class="bg-red-700 hover:bg-red-600 rounded-lg text-lg font-bold p-2 text-white w-1/2 hover:cursor-pointer"
              @click="emitConfirm"
            >
              {{ t('actions.delete') }}
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
