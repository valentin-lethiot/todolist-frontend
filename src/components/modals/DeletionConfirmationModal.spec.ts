import { describe, beforeEach, expect, it } from 'vitest'
import DeletionConfirmationModal from './DeletionConfirmationModal.vue'
import { mount, VueWrapper } from '@vue/test-utils'

describe('DeletionConfirmationModal', () => {
  let wrapper: VueWrapper

  const defaultProps = {
    isOpen: true,
  }

  beforeEach(() => {
    wrapper = mount(DeletionConfirmationModal, {
      props: defaultProps,
    })
  })

  describe('render', () => {
    it('should render the modal', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('events', () => {
    it('should emit close when clicking cancel button', async () => {
      await wrapper.find('#cancelButton').trigger('click')
      expect(wrapper.emitted('close')).toBeTruthy()
    })

    it('should emit confirm when clicking delete button', async () => {
      await wrapper.find('#confirmButton').trigger('click')
      expect(wrapper.emitted('confirm')).toBeTruthy()
    })
  })
})
