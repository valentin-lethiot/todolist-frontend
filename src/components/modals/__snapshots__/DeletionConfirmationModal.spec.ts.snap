// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeletionConfirmationModal > render > should render the modal 1`] = `
"<div class="absolute w-screen h-screen top-0 left-0 bg-gray-900/40 z-60">
  <div class="w-full h-full flex flex-col justify-center items-center p-8">
    <div class="bg-gray-800 rounded-lg p-4 shadow-xl w-1/2">
      <div class="flex flex-col w-full h-full gap-4">
        <div class="h-1/2 text-xl font-bold flex justify-between items-start">
          <p>Êtes-vous sûr de vouloir supprimer cette tâche ?</p>
        </div>
        <div class="h-1/2 w-full flex gap-4 items-center justify-end"><button id="cancelButton" class="bg-gray-700 hover:bg-gray-600 rounded-lg text-lg font-bold p-2 text-white w-1/2 hover:cursor-pointer">Annuler</button><button id="confirmButton" class="bg-red-700 hover:bg-red-600 rounded-lg text-lg font-bold p-2 text-white w-1/2 hover:cursor-pointer">Supprimer</button></div>
      </div>
    </div>
  </div>
</div>"
`;
